spec/compiler/semantic/proc_spec.cr

```diff
@@ -916,4 +916,28 @@ describe "Semantic: proc" do
       foo(proc: ->{ 1 })
       )) { proc_of(nil_type) }
   end
+
+  it "errors when using macro as proc value (top-level) (#7465)" do
+    assert_error %(
+      macro bar
+      end
+
+      ->bar
+      ),
+      "'bar' is a macro and macros can't be used in proc pointers"
```

Comment:
Can we make the error message `undefined method 'bar' for top-level` as it would be if `bar` was not defined or a local var? That's the error here: there is no method of that name. Whether there a synonymous macro might just be a minor coincidence.

It doesn't hurt to keep the current message as an additional hint though.

Reply:
Yeah, that probably makes sense. I'll change it later.